torch/_inductor/dependencies.py

```diff
@@ -475,6 +475,10 @@ def indirect_indexing(self, index_var, size, check=True) -> sympy.Symbol:
     def frexp(self, x):
         return (None,) * 2
 
+    def scan(self, dtypes, combine_fn, values, inits):
+        self.inner_fn(dtypes, combine_fn, values, inits)
```

Comment:
What is `inner_fn` here?

Reply:
I wanted to run the function returned by `__getattr__` and this was just a hack to get that. Let me rewrite it better.

Reply:
I see, that's a neat trick but a bit confusing. I also don't think it's necessary. The inputs shouldn't ever be `sympy.Expr` or `sympy.logic.boolalg.Boolean`.